jersey-pac4j + Payara 5 not working

Hi, I migrate from Payara 4 to Payara 5 and updates pac4j libraries from version 1.9.x to the last one 3.x.x .I made all the changes neccesaries to run the project but when i tried i get this warning:

> The following warnings have been detected: WARNING: HK2 service reification failed for [org.pac4j.jax.rs.jersey.features.Pac4JValueFactoryProvider$Pac4JProfileValueFactoryProvider] with an exception:
> MultiException stack 1 of 4
> java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/glassfish/jersey/server/model/Parameter

The stack trace: 

```
[#|2018-08-22T10:58:05.431-0300|INFORMACIÓN|Payara 5.182||_ThreadID=53;_ThreadName=admin-thread-pool::admin-listener(3);_TimeMillis=1534946285431;_LevelValue=800;|
  LDAP profile seba: #LdapProfileService# | connectionFactory: [org.ldaptive.DefaultConnectionFactory@403108426::provider=org.ldaptive.provider.jndi.JndiProvider@40f9bc9d, config=[org.ldaptive.ConnectionConfig@2139092744::ldapUrl=ldap://ldapdev.ucasal.edu.ar:389, connectTimeout=PT5S, responseTimeout=PT15S, sslConfig=null, useSSL=false, useStartTLS=false, connectionInitializer=[org.ldaptive.BindConnectionInitializer@1929308613::bindDn=uid=weblogic,ou=users,ou=access-list,dc=ucasal,dc=edu,dc=ar, bindSaslConfig=null, bindControls=null], connectionStrategy=org.ldaptive.DefaultConnectionStrategy@5dc5c172]] | ldapAuthenticator: [ar.edu.ucasal.sagbackend.security.ConfigSecurity$1@1347781379::dnResolver=[org.ldaptive.auth.FormatDnResolver@1351805366::formatString=uid=%s,ou=people,dc=ucasal,dc=edu,dc=ar, formatArgs=null, escapeUser=true], authenticationHandler=[org.ldaptive.auth.PooledBindAuthenticationHandler@1428652704::factory=[org.ldaptive.pool.PooledConnectionFactory@647628583::pool=[org.ldaptive.pool.BlockingConnectionPool@1521206636::name=null, poolConfig=[org.ldaptive.pool.PoolConfig@489904342::minPoolSize=5, maxPoolSize=50, validateOnCheckIn=true, validateOnCheckOut=true, validatePeriodically=false, validatePeriod=PT30M, validateTimeout=PT5S], activator=null, passivator=null, validator=[org.ldaptive.pool.SearchValidator@1961425746::searchRequest=[org.ldaptive.SearchRequest@-1856859137::baseDn=, searchFilter=[org.ldaptive.SearchFilter@1642584434::filter=(objectClass=*), parameters={}], returnAttributes=[1.1], searchScope=OBJECT, timeLimit=PT0S, sizeLimit=1, derefAliases=null, typesOnly=false, binaryAttributes=null, sortBehavior=UNORDERED, searchEntryHandlers=null, searchReferenceHandlers=null, controls=null, referralHandler=null, intermediateResponseHandlers=null]] pruneStrategy=[org.ldaptive.pool.IdlePruneStrategy@37579223::prunePeriod=PT5M, idleTime=PT10M], connectOnCreate=true, connectionFactory=[org.ldaptive.DefaultConnectionFactory@403108426::provider=org.ldaptive.provider.jndi.JndiProvider@40f9bc9d, config=[org.ldaptive.ConnectionConfig@2139092744::ldapUrl=ldap://ldapdev.ucasal.edu.ar:389, connectTimeout=PT5S, responseTimeout=PT15S, sslConfig=null, useSSL=false, useStartTLS=false, connectionInitializer=[org.ldaptive.BindConnectionInitializer@1929308613::bindDn=uid=weblogic,ou=users,ou=access-list,dc=ucasal,dc=edu,dc=ar, bindSaslConfig=null, bindControls=null], connectionStrategy=org.ldaptive.DefaultConnectionStrategy@5dc5c172]], initialized=true, availableCount=5, activeCount=0]], saslConfig=null, controls=null], entryResolver=null, returnAttributes=null, authenticationResponseHandlers=null] | usersDn: ou=people,dc=ucasal,dc=edu,dc=ar | idAttribute: id | attributes: cn,mail,memberOf | profileDefinition: null ||#]
[#|2018-08-22T10:58:06.816-0300|ADVERTENCIA|Payara 5.182|org.glassfish.jersey.internal.Errors|_ThreadID=53;_ThreadName=admin-thread-pool::admin-listener(3);_TimeMillis=1534946286816;_LevelValue=900;|
  The following warnings have been detected: WARNING: HK2 service reification failed for [org.pac4j.jax.rs.jersey.features.Pac4JValueFactoryProvider$Pac4JProfileValueFactoryProvider] with an exception:
MultiException stack 1 of 4
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/glassfish/jersey/server/model/Parameter
	at org.glassfish.hk2.utilities.cache.LRUHybridCache.compute(LRUHybridCache.java:315)
	at org.glassfish.hk2.utilities.reflection.internal.ClassReflectionHelperImpl.getAllMethods(ClassReflectionHelperImpl.java:108)
	at org.jvnet.hk2.internal.Utilities.findInitializerMethods(Utilities.java:1424)
	at org.jvnet.hk2.internal.DefaultClassAnalyzer.getInitializerMethods(DefaultClassAnalyzer.java:107)
	at org.glassfish.jersey.inject.hk2.JerseyClassAnalyzer.getInitializerMethods(JerseyClassAnalyzer.java:242)
	at org.jvnet.hk2.internal.Utilities.getInitMethods(Utilities.java:220)
	at org.jvnet.hk2.internal.ClazzCreator.initialize(ClazzCreator.java:145)
	at org.jvnet.hk2.internal.ClazzCreator.initialize(ClazzCreator.java:180)
	at org.jvnet.hk2.internal.SystemDescriptor.internalReify(SystemDescriptor.java:740)
	at org.jvnet.hk2.internal.SystemDescriptor.reify(SystemDescriptor.java:694)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.reifyDescriptor(ServiceLocatorImpl.java:464)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.narrow(ServiceLocatorImpl.java:2310)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.access$1200(ServiceLocatorImpl.java:128)
	at org.jvnet.hk2.internal.ServiceLocatorImpl$10.compute(ServiceLocatorImpl.java:1395)
	at org.jvnet.hk2.internal.ServiceLocatorImpl$10.compute(ServiceLocatorImpl.java:1390)
	at org.glassfish.hk2.utilities.cache.internal.WeakCARCacheImpl.compute(WeakCARCacheImpl.java:128)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.internalGetAllServiceHandles(ServiceLocatorImpl.java:1452)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getAllServices(ServiceLocatorImpl.java:818)
	at org.glassfish.jersey.inject.hk2.AbstractHk2InjectionManager.getAllInstances(AbstractHk2InjectionManager.java:194)
	at org.glassfish.jersey.inject.hk2.ImmediateHk2InjectionManager.getAllInstances(ImmediateHk2InjectionManager.java:54)
	at org.glassfish.jersey.server.internal.inject.ValueParamProviderConfigurator.postInit(ValueParamProviderConfigurator.java:181)
	at org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$2(ApplicationHandler.java:373)
	at org.glassfish.jersey.server.ApplicationHandler$$Lambda$131/692175296.accept(Unknown Source)
	at java.util.Arrays$ArrayList.forEach(Arrays.java:3880)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:373)
	at org.glassfish.jersey.server.ApplicationHandler.lambda$initialize$1(ApplicationHandler.java:317)
	at org.glassfish.jersey.server.ApplicationHandler$$Lambda$95/1906739785.call(Unknown Source)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
	at org.glassfish.jersey.internal.Errors.processWithException(Errors.java:256)
	at org.glassfish.jersey.server.ApplicationHandler.initialize(ApplicationHandler.java:316)
	at org.glassfish.jersey.server.ApplicationHandler.<init>(ApplicationHandler.java:283)
	at org.glassfish.jersey.servlet.WebComponent.<init>(WebComponent.java:335)
	at org.glassfish.jersey.servlet.ServletContainer.init(ServletContainer.java:178)
	at org.glassfish.jersey.servlet.ServletContainer.init(ServletContainer.java:370)
	at javax.servlet.GenericServlet.init(GenericServlet.java:244)
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1515)
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1323)
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:5494)
	at org.apache.catalina.core.StandardContext.start(StandardContext.java:5739)
	at com.sun.enterprise.web.WebModule.start(WebModule.java:618)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:956)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:939)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:684)
	at com.sun.enterprise.web.WebContainer.loadWebModule(WebContainer.java:2105)
	at com.sun.enterprise.web.WebContainer.loadWebModule(WebContainer.java:1750)
	at com.sun.enterprise.web.WebApplication.start(WebApplication.java:107)
	at org.glassfish.internal.data.EngineRef.start(EngineRef.java:123)
	at org.glassfish.internal.data.ModuleInfo.start(ModuleInfo.java:294)
	at org.glassfish.internal.data.ApplicationInfo.start(ApplicationInfo.java:357)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.initialize(ApplicationLifecycle.java:541)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:542)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:549)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:545)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:544)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:575)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:567)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:566)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1475)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:111)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1857)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1733)
	at com.sun.enterprise.v3.admin.AdminAdapter.doCommand(AdminAdapter.java:564)
	at com.sun.enterprise.v3.admin.AdminAdapter.onMissingResource(AdminAdapter.java:251)
	at org.glassfish.grizzly.http.server.StaticHttpHandlerBase.service(StaticHttpHandlerBase.java:166)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:516)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:213)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/glassfish/jersey/server/model/Parameter
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.glassfish.hk2.utilities.cache.LRUHybridCache$OriginThreadAwareFuture.get(LRUHybridCache.java:164)
	at org.glassfish.hk2.utilities.cache.LRUHybridCache.compute(LRUHybridCache.java:303)
	... 87 more
```


I use the ldap auhtenticator and its authenticates correctly:  DEBUG o.p.core.engine.DefaultSecurityLogic - authenticated and authorized -> grant access

but when I inject the profile with @Pac4JProfile LdapProfile profile i always get null.

--- a/jersey/src/main/java/org/pac4j/jax/rs/jersey/features/Pac4JValueFactoryProvider.java
+++ b/jersey/src/main/java/org/pac4j/jax/rs/jersey/features/Pac4JValueFactoryProvider.java
@@ -179,9 +179,9 @@ public class Pac4JValueFactoryProvider {
 
         /**
          * Use this if you want to return a dynamically supplied {@link CommonProfile} (or none with <code>null</code>).
-         * 
+         *
          * Note that it won't mock the profile coming out of {@link ProfileManager}!
-         * 
+         *
          * @param profile
          *            a profile supplier, can return <code>null</code>.
          */
@@ -196,19 +196,18 @@ public class Pac4JValueFactoryProvider {
 
             if(manager == null){
                 bind(DefaultProfileManagerFactoryBuilder.class)
-                    .to(ProfileManagerFactoryBuilder.class)
-                ;
+                    .to(ProfileManagerFactoryBuilder.class).in(Singleton.class);
             } else {
-                bind(manager).to(ProfileManagerFactoryBuilder.class);
+                bind(manager).to(ProfileManagerFactoryBuilder.class).in(Singleton.class);
             }
 
-            bind(Pac4JProfileValueFactoryProvider.class).to(ValueParamProvider.class).in(Singleton.class);
+            bindAsContract(Pac4JProfileValueFactoryProvider.class).to(ValueParamProvider.class).in(Singleton.class);
 
-            bind(ProfileInjectionResolver.class)
+            bindAsContract(ProfileInjectionResolver.class)
                 .to(new GenericType<InjectionResolver<Pac4JProfile>>(){})
                 .in(Singleton.class);
 
-            bind(ProfileManagerInjectionResolver.class)
+            bindAsContract(ProfileManagerInjectionResolver.class)
                 .to(new GenericType<InjectionResolver<Pac4JProfileManager>>(){})
                 .in(Singleton.class);
         }

--- a/jersey/src/main/java/org/pac4j/jax/rs/jersey/features/Pac4JValueFactoryProvider.java
+++ b/jersey/src/main/java/org/pac4j/jax/rs/jersey/features/Pac4JValueFactoryProvider.java
@@ -198,7 +198,7 @@ public class Pac4JValueFactoryProvider {
                 bind(DefaultProfileManagerFactoryBuilder.class)
                     .to(ProfileManagerFactoryBuilder.class).in(Singleton.class);
             } else {
-                bind(manager).to(ProfileManagerFactoryBuilder.class).in(Singleton.class);
+                bind(manager).to(ProfileManagerFactoryBuilder.class);
             }
 
             bindAsContract(Pac4JProfileValueFactoryProvider.class).to(ValueParamProvider.class).in(Singleton.class);

